azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/ConfigMojo.java

```diff
@@ -0,0 +1,340 @@
+        if (!settings.isInteractiveMode()) {
+            throw new MojoFailureException("The goal 'config' must be run at interactive mode.");
+        }
+
+        if (!Utils.isPomPackagingProject(this.project) && !Utils.isJarPackagingProject(this.project)) {
+            throw new MojoFailureException(String.format("The project (%s) with packaging %s is not supported for azure spring cloud service.",
+                    this.project.getName(), this.project.getPackaging()));
+        }
+        if (isProjectConfigured(this.project)) {
+            throw new MojoFailureException(String.format("Project (%s) is already configured.", this.project.getName()));
+        }
+
+        if (this.deployment == null) {
+            // make sure deployment is not null
+            this.deployment = new Deployment();
+        }
+
+        final boolean isRunningInParent = Utils.isPomPackagingProject(this.project);
+
+        try {
+            if (isRunningInParent) {
+                if (full) {
+                    throw new MojoFailureException("The \"full\" mode is not supported at parent folder.");
+                }
+                initializeProjects();
+            }
+            initializeCredentials();
+            selectAppCluster();
+
+            if (isRunningInParent) {
+                configurePublicParent();
+                confirmAndSaveParent();
+            } else {
+                configureAppName();
+                configurePublic();
+                if (full) {
+                    configureInstanceCount();
+                    configureCpu();
+                    configureMemory();
+                    configureJvmOptions();
+                } else {
+                    this.deployment.withCpu(1).withMemoryInGB(2).withInstanceCount(1).withJvmOptions("-Xmx1G");
+                }
+                confirmAndSave();
+            }
+
+            this.prompt.close();
+        } catch (IOException | NoResourcesAvailableException | InvalidConfigurationException e) {
+            throw new MojoFailureException(e.getMessage());
+        }
+    }
+
+    private void initializeProjects() throws MojoFailureException, IOException {
+        final List<MavenProject> allProjects = session.getAllProjects().stream().filter(Utils::isJarPackagingProject).collect(Collectors.toList());
+        final List<String> configuredProjects = new ArrayList<>();
+        for (final MavenProject proj : allProjects) {
+            if (isProjectConfigured(proj)) {
+                configuredProjects.add(proj.getName());
+            } else {
+                targetProjects.add(proj);
+            }
+        }
+        if (configuredProjects.size() > 0) {
+            getLog().warn(String.format("Project (%s) %s already configured.",
+                    TextUtils.yellow(configuredProjects.stream().collect(Collectors.joining(", "))), configuredProjects.size() > 1 ? "are" : "is"));
+        }
+
+        if (targetProjects.size() == 0) {
```

Comment:
Seems that `config` will still do `configurePublicParent();` even if all child projects has been configured, is it by design?

Reply:
done